### ISSUE TITLE
Question - How to authenticate against custom AAD protected api's?

### ISSUE THREAD

**Issue by btastic** [Clancey/simple_auth#50]
Hey there,

we have a few C# API's that are protected by our azure active directory. In our typescript frontends, we use adal.js to authenticate with the clientId and the proper redirectUri. How can I do the same thing with your library? Specifically, I have problems figuring out what to put in each parameter.

```dart
final simpleAuth.AzureADApi azureApi = simpleAuth.AzureADApi(
      '<name of the tool>',
      '<clientId>',
      'https://login.microsoftonline.com/<our tenant>/oauth2/token',
      '<what do I put here?>',
      'https://login.microsoftonline.com/<our tenant>/oauth2/authorize',
      redirectUrl: 'msal<generated by app registration portal>://auth');
```

Can you point me in the right direction?

Thanks!

**Comment by Clancey** [comment]
There is a sample: https://github.com/Clancey/simple_auth/blob/master/simple_auth_flutter_example/lib/api_definitions/azureAdTestApi.simple_auth.dart

**Comment by btastic** [comment]
Wow, thanks for your quick response.

I worked through every example I found on github (searching for open source code, that is referencing your library) but none of them show the real deal, or how to get the proper values in there. Everywhere just placeholders (which I get, due to sensitive data). The problem that I have is, that I don't know where I should my client_secret get from. I don't have that on my azure app registration portal.

Also I think I have a valid starting point in my sample code I have posted above. I just need to fill the correct values I guess?
